File: 230.java
import java.util.*;

class Solution {
    public int kthSmallest(TreeNode root, int k) {
        List<Integer> infixList = new ArrayList<>(k);
        infix(root, infixList, k);
        return infixList.get(k - 1);
    }

    private void infix(TreeNode root, List<Integer> infixList, int k) {
        if (root == null) {
            return;
        }
        infix(root.left, infixList, k);
        infixList.add(root.val);
        if (infixList.size() == k) {
            return;
        }
        infix(root.right, infixList, k);
    }
}

File: Python/230.py
from typing import Optional


# Definition for a binary tree node.
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right


class Solution:
    def kthSmallest(self, root: Optional[TreeNode], k: int) -> int:
        count, result = 0, 0

        def helper(root):
            nonlocal count
            nonlocal result
            if not root:
                return
            helper(root.left)
            count += 1
            if count == k:
                result = root.val
                return
            helper(root.right)

        helper(root)
        return result
